README.rst

```diff
@@ -1,11 +1,11 @@
 cloud-pubsub-logging-python
```

Comment:
It seems like the "How to Use" section should be split into two parts: one for sync and one for async. Otherwise, if you're advocating for just async, maybe spell that out somewhere.